pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java

```diff
@@ -151,18 +153,32 @@ public void start() {
         runnerThread.start();
     }
 
-    public abstract V extractValue(ConsumerRecord<String, byte[]> record);
+    public Object extractValue(ConsumerRecord<Object, Object> consumerRecord) {
+        return consumerRecord.value();
+    }
 
+    public Optional<String> extractKey(ConsumerRecord<Object, Object> consumerRecord) {
+        // we are currently supporting only String keys
+        return Optional.ofNullable((String) consumerRecord.key());
+    }
+
+    public abstract Schema<V> extractSchema(ConsumerRecord<Object, Object> consumerRecord);
+
+    @Slf4j
     static private class KafkaRecord<V> implements Record<V> {
```

Comment:
(see previous comment about the type parameter to KafkaRecord class)

Reply:
(see previous answer)